RTree.h

```diff
@@ -114,6 +114,12 @@ class RTree
   /// \return Returns the number of entries found
   size_t NNSearch(const ELEMTYPE a_min[NUMDIMS], const ELEMTYPE a_max[NUMDIMS], std::function<bool(const DATATYPE&, ELEMTYPE)> callback) const;
 
+  /// Aggregate nodes hierarchically
+  /// \param convert a function to convert the data type to the aggregation type
+  /// \param combine a function to combine two aggregation types
+  /// \param progress a function to check if the aggregation should continue
+  template <typename T>
+  T Aggregate(std::function<T(const DATATYPE &a1)> convert, std::function<T(T &a1, T &a2)> combine, std::function<bool()> progress);
```

Comment:
Could you please provide an example for the `progress` function? I imagined it depends on the current data to determine if to continue, but it does not get any parameters. Is it used to halt the execution from another thread?